core/test/utils/assertions.hpp

```diff
@@ -91,12 +95,14 @@ void print_componentwise_error(Ostream &os, const MatrixData1 &first,
         for (size_type col = 0; col < first.size[1]; ++col) {
             auto r = get_next_value(first_it, end(first.nonzeros), row, col);
             auto e = get_next_value(second_it, end(second.nonzeros), row, col);
-            auto m =
-                max(static_cast<real_vt>(abs(r)), static_cast<real_vt>(abs(e)));
+            auto rr = static_cast<real_vt>(std::real(r));
+            auto re = static_cast<real_vt>(std::real(e));
+            auto m = std::max(static_cast<real_vt>(abs(r)),
+                              static_cast<real_vt>(abs(e)));
```

Comment:
Here we again only use the real part and cast to `real_vt`. We could probably simplify this to 
```suggestion
            auto m = std::max<double>(abs(r), abs(e));
```